src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -119,14 +133,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Protected Overrides Function ExpressionKind() As OperationKind
             Dim rightBinary As BoundBinaryOperator = TryCast(Me.Right, BoundBinaryOperator)
             If rightBinary IsNot Nothing Then
-                If TypeOf rightBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If rightBinary.Left.Kind = BoundKind.CompoundAssignmentTargetPlaceholder Then
                     Return OperationKind.CompoundAssignmentExpression
                 End If
             End If
 
             Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = TryCast(Me.Right, BoundUserDefinedBinaryOperator)
             If rightOperatorBinary IsNot Nothing Then
-                If TypeOf rightOperatorBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If DirectCast(rightOperatorBinary, IBinaryOperatorExpression).LeftOperand.Kind = OperationKind.PlaceholderExpression Then
```

Comment:
@JohnHamby If my observation about OperationKind is correct, before making another round of changes here, let's discuss what it might look like offline.